oak_docker_linux_init/prepare_docker_launcher_initramfs.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+#
+# Prepares an initramfs file by combining the following:
+#    - Alpine Linux's minirootfs.
+#    - Necessary drivers copied from an Alpine Linux distribution.
+#    - Filesystem extracted from the given docker image.
+#
+#
+
+ # exit when any command fails
+set -e
+
+readonly SCRIPT_DIR=$(dirname "$0")
+
+print_usage_and_exit() {
+  echo "Usage:
+  ${0} [-h] \\
+       -k <uncompressed-kernel> \\
+       -i <alpine-initiramfs>   \\
+       -m <alpine-minirootfs>   \\
+       -o <output-initramfs>    \\
+       -d <docker-image>
+
+Creates an initramfs with minirootfs+networking support as docker launcher.
+
+Options:
+  -k      Uncompressed kernel with which we use the drivers.
+  -i      Initramfs extracted from an Alpine iso distribution.
+  -m      Alpine's minirootfs tarball.
+  -d      Docker image that should be launched.
+  -o      Output initramfs file.
+  -h      Print this help message and exit."
+  exit 0
+}
+
+while getopts "hk:i:m:d:o:" opt; do
+  case $opt in
+    h)
+      print_usage_and_exit;;
+    k)
+      readonly LINUX_KERNEL="${OPTARG}";;
+    i)
+      readonly ALPINE_INITRAMFS="${OPTARG}";;
+    m)
+      readonly ALPINE_MINIROOTFS_TAR="${OPTARG}";;
+    d)
+      readonly DOCKER_IMAGE="${OPTARG}";;
+    o)
+      readonly OUTPUT_INITRAMFS="${OPTARG}";;
+    *)
+      echo "Invalid argument: ${OPTARG}"
+      exit 1;;
+  esac
+done
+
+if [ -z "${LINUX_KERNEL}" ]; then
+  echo "Missing required option: -k <uncompressed-kernel>"
+  exit 1
+fi
+
+if [ -z "${ALPINE_INITRAMFS}" ]; then
+  echo "Missing required option: -i <alpine-initramfs>"
+  exit 1
+fi
+
+if [ -z "${OUTPUT_INITRAMFS}" ]; then
+  echo "Missing required option: -o <output-initramfs>"
+  exit 1
+fi
+
+if [ -z "${ALPINE_MINIROOTFS_TAR}" ]; then
+  echo "Missing required option: -m <alpine-minirootfs>"
+  exit 1
+fi
+
+# Create a temporary workspace directory.
+readonly SCRATCH_DIR=$(mktemp -d)
+echo "[INFO] Scratch directory is ${SCRATCH_DIR}"
+# Make a directory for the initramfs.
+readonly RAMDIR=$(mktemp -d)
+echo "[INFO] Ramdisk staging directory is ${RAMDIR}"
+
+# Extract the linux version string from the kernel image.
+readonly LINUX_KERNEL_VERSION="$(strings "${LINUX_KERNEL}" | \
+                                 grep "Linux version" | cut -d\  -f3)"
+echo "[INFO] Linux kernel version is ${LINUX_KERNEL_VERSION}"
+
+
+# Extract Alpine's initramfs to a scratch directory
```

Comment:
nit: for consistency here and elsewhere, comments should end with a `.`

Reply:
done.